controllers/apply/reaching-communities/__tests__/processor.test.js

```diff
@@ -0,0 +1,62 @@
+/* eslint-env jest */
+'use strict';
+const nodemailer = require('nodemailer');
+
+const processor = require('../processor');
+const form = require('../form-model');
+const { flattenFormData, stepsWithValues } = require('../../helpers');
+
+const mockFormData = {
+    'step-1': {
+        'your-idea': 'Test idea'
+    },
+    'step-2': {
+        location: 'South West',
+        'project-location': 'London'
+    },
+    'step-3': {
+        'organisation-name': 'Test Organisation',
+        'additional-organisations': ''
+    },
+    'step-4': {
+        'first-name': 'Example',
+        'last-name': 'Person',
+        email: 'example@example.com',
+        'phone-number': '03454102030'
+    }
+};
+
+function cleanMailForSnaphot(info) {
+    info.message = info.message
+        .toString()
+        .replace(/Message-ID: .*\n/, '')
+        .replace(/Date: .*\n/, '')
+        .replace(/ boundary=.*\n/, '')
+        .replace(/----.*\n/gm, '');
+    return info;
+}
+
+describe('processor', () => {
+    it('should work', async () => {
+        const transport = nodemailer.createTransport({
+            streamTransport: true,
+            newline: 'unix',
+            buffer: true
+        });
+
+        const result = await processor(
+            {
+                form: form,
+                data: flattenFormData(mockFormData),
+                stepsWithValues: stepsWithValues(form.steps, mockFormData)
+            },
+            transport
```

Comment:
In order for this approach to work we have to pass a custom transport down through the different functions. This is the most awkward feeling part as we have to pass it down three levels. There might be a way around this if we pull the transport part up but at the expense of more boilerplate elsewhere. Conflicted.